internal/provider/data_source_aws_network_peering.go

```diff
@@ -127,25 +129,42 @@ func dataSourceAwsNetworkPeeringRead(ctx context.Context, d *schema.ResourceData
 	}
 
 	// Skip waiting.
-	if !waitForActive {
+	if !waitForActive || peering.State == models.HashicorpCloudNetwork20200907PeeringStateACTIVE {
 		return nil
 	}
 
 	// If it's in a state where it could later become ACTIVE, wait.
-	for _, state := range clients.WaitForPeeringToBeActivePendingStates {
+	waitState := false
```

Comment:
I'm a bit torn on DRY'ing this into the Waiters because of the shenanigans I'm trying to do here where:
- if it's a terminal state (FAILED/DELETING) at the _start_ of the Read, throw a Warning
- if it's a terminal state after trying to _wait_ in the Read, throw an Error

Also this is a specific (hack?) for the Peering data resources that I wouldn't want to add to `WaitForPeeringToBeAccepted` for example because we always want it to throw an Error there: https://github.com/hashicorp/terraform-provider-hcp/blob/441597cc026ffc6ebd02bd149a953f77fbfd110c/internal/provider/resource_hvn_peering_connection.go#L156

I'm not opposed necessarily, but because of two reasons above I left it within the data_resource_* Reads

Reply:
Hmm I see, that does make sense. In the interest of getting this in, we can leave it like this for now, and revisit whenever we add another peering resource.